app/controllers/openid_application_controller.rb

```diff
@@ -35,8 +36,11 @@ def authenticate_token
     return false if token.blank?
 
     # issued for a client vs a user
-    if token.client_credentials_token?
+    if token.client_credentials_token? || ssoi_token?
```

Comment:
Would it be worth breaking these two conditionals apart for clarity?

The `launch/patient` logic conditionally applies only for client credentials.
The `launch` logic conditionally applies only for internal users.